### PR TITLE
Enhance ad server kubernetes level tests with member server setup

### PR DESCRIPTION
**Pull request by obnoxxx**

### CODE REVIEW

**Comment by obnoxxx** [comment]
@phlogistonjohn - FYI, WIP PR with kubernetes level ad-server and ad-member test.

File: tests/test-remove-ad-server.sh [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+SCRIPT_DIR="$(cd "$(dirname "${0}")" && pwd)"
+
+source "${SCRIPT_DIR}/common.sh"
+
+if [ ${KEEP} -eq 1 ]; then
+	echo "keeping ad server deployment (KEEP=1)"
+	exit 0
+fi
+
+echo "removing ad server deployment again..."
```
Comment (phlogistonjohn) [inline tests/test-remove-ad-server.sh:12]: Very nitpicky, but the word "again" here strikes me as unnecessary.

File: tests/test-remove-ad-server.sh [hunk 2]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+SCRIPT_DIR="$(cd "$(dirname "${0}")" && pwd)"
+
+source "${SCRIPT_DIR}/common.sh"
+
+if [ ${KEEP} -eq 1 ]; then
```
Comment (phlogistonjohn) [inline tests/test-remove-ad-server.sh:7]: Should this be named something more specific like KEEP_AD or KEEP_AD_DEPLOYMENT?

File: tests/test-samba-ad-server-kubernetes.sh [hunk 3]
```diff
@@ -10,6 +10,14 @@ kubectl exec "${podname}" -- samba-tool domain info 127.0.0.1
 [ $? -eq 0 ] || _error "Error listing domain info"
 echo
 
+source ${SCRIPT_DIR}/test-deploy-ad-member.sh
+
+kubectl exec "${podname}" -c "smb" -- smbclient -N -L 127.0.0.1
+kubectl exec "${podname}" -c "winbind" -- wbinfo -t
+kubectl exec "${podname}" -c "smb" -- wbinfo -t
```
Comment (phlogistonjohn) [inline tests/test-samba-ad-server-kubernetes.sh:17]: I was momentarily confused by two "wbinfo -t" but then I realized it makes sense as the container running smbd needs to communicate with winbind and `wbinfo -t` can demonstrate that. So, perhaps a comment will save future readers like me from that confusion.

**Review by phlogistonjohn — COMMENTED** [review]
General +1 to the gist of things here. But as it's still in draft I will look in more detail later.

**Comment by phlogistonjohn** [comment]
While reading this I also realized we should do something about continuing to use "my" domain and filed issue #39

File: tests/test-remove-ad-server.sh (hunk 1 above)
Comment (obnoxxx) [reply tests/test-remove-ad-server.sh:12]: right, I thought I had changed that...

File: tests/test-remove-ad-server.sh (hunk 2 above)
Comment (obnoxxx) [reply tests/test-remove-ad-server.sh:7]: Well, I was thinking about this.
The current patch uses `$KEEP` in both `test-remove-ad-server.sh` and `test-remove-ad-member.sh`.
I could introduce two separate variables...

🤔

File: tests/test-samba-ad-server-kubernetes.sh (hunk 3 above)
Comment (obnoxxx) [reply tests/test-samba-ad-server-kubernetes.sh:17]: exactly, that's the point. :-)
Will add a comment.

**Comment by obnoxxx** [comment]
> While reading this I also realized we should do something about continuing to use "my" domain and filed issue #39

Right. Let me integrate into this PR a patch to first change the domain to something else.

File: tests/test-samba-ad-server-kubernetes.sh (hunk 3 above)
Comment (obnoxxx) [reply tests/test-samba-ad-server-kubernetes.sh:19]: done

File: tests/test-remove-ad-server.sh (hunk 1 above)
Comment (obnoxxx) [reply tests/test-remove-ad-server.sh]: done

File: tests/test-remove-ad-server.sh (hunk 2 above)
Comment (obnoxxx) [reply tests/test-remove-ad-server.sh:7]: kept is as one var for now

**Comment by obnoxxx** [comment]
updated and polished the PR.
Based on #40 now

**Comment by obnoxxx** [comment]
@phlogistonjohn  -  updated the test to test the locally built image instead of the one from quay

**Review by phlogistonjohn — APPROVED** [review]
Looks good. Since nothing here is set in stone I don't think its worth nitpicking anything else. :-)

(Well, I did notice that one commit messages says "kubernets"... but I will approve anyway and leave it to you to either fix or not worry about a minor typo)

**Comment by obnoxxx** [comment]
Ah, you don't know this kubernets? ... 😁 
I think I don't care.

**Pull request merged**